test/unit_test/runtime/fmt_test.cpp

```diff
@@ -296,7 +296,7 @@ TEST_CASE("operator<< on a quantity", "[text][ostream][fmt]")
 
       SECTION("iostream")
       {
-        CHECK(stream.str() == "8 [1 x 10⁻²]m³");
+        CHECK(stream.str() == "8 [1 \u00D7 10⁻²]m³");
```

Comment:
Please for tests just paste the sign (like we did for superscripts) rather printing its code. It is easier to understand what the correct output should look like.

Reply:
you mean for the test here, or for the code where the string is constructed, or both? 

I agree with the test, but the string gen code, I am not sure about?

Reply:
what is the encoding of our src files? UTF8? There is no BOM?

I have to say, it feels like a pandorra's box to me. Just to make a test more readable (if it ever fails), we should start having UTF8 / unicode encoded source files? I have spent years battling encodings and I don't like them. 

Or am I misunderstanding something?

Reply:
I guess what I am saying is: If you really want me to put "\u00d7" in that test file as a single character rather then a unicode literal, then you need to tell me how to do it. 

I am not going to do by "pasting", because that is a total black box. 

I am going to switch my editor to hexadecimal mode and enter the actual bytes in that string literal. So what should those bytes be?

And should there be anything at the top of the file to let the compiler know what we are doing? (ie a BOM)

Reply:
Well, I did not check if there is BOM or not in a file but my IDE (VS Code) claims the *.cpp file is encoded in UTF-8 encoding.

Reply:
There is no BOM. MS products are very "presumptious and automatic black-box-like" in their treatment of encoding systems. 

I am happy to put whatever you want there, but I don't want to be reponsible for a "paste" operation. I want the hex-bytes. 

(BTW: I am using emacs on linux in a terminal window, and that, as well as prior experience is why I am not going to rely on paste. ). 

I have previously had lots of trouble with people "pasting from some MS application" because no one ultimately knows what is in there, what it is supposed to be, or how the compiler will interpret it. 

I have done some searching and I believe the entire subject of non-ASCII in c++ source code files remains totally unresolved and is "implementation dependent".

Reply:
Here are the options: http://www.fileformat.info/info/unicode/char/d7/index.htm

Reply:
BOM is [neither required nor recommended](https://en.wikipedia.org/wiki/Byte_order_mark#UTF-8) for UTF-8.

Reply:
That may be. But what does the C++ standard say about anything but ASCII in source code? That's what matters here. 

Like I say, I will put anything there you want me to, but please give me the hex bytes.

Reply:
You need "UTF-8 (hex)", i.e. `0xC3 0x97`. Yes, it is differs from code point, that's OK.

[Here](http://utf8everywhere.org/#faq.literal) are few more recommendations about Unicode literals and workarounds for MSVC.

Reply:
I know what the UTF8 hex bytes are for that character. They are on that page I linked to. 

The question is, what do we want in the source code, which is standards compliant and portable? 

Please note that in the place where we are construct the string (that we are testing for) we are using "\u00d7". You seem to be suggesting that you would like me to change the test such it is testing for `\xc3\x97" (just inserted into the string literal as a character rather than  the \x or \u escape sequence)? And you are hoping that somewhere (??) there will be a magic conversion in a portable standards compliant way between \x00\xd7 and \xc3\x97? 

I think I will leave it up to you and @mpusz You can just accept the pull request and change it to whatever you want. My experience says 

"This is a complicated topic. The standard is not clear and it is "implementation dependent" (to my knowledge). So best to use the escape sequences provided by the standard". 

If you want to handle it differently, feel free.

Reply:
You are making good points! I came from "UTF-8 everywhere" background and my perception is a bit distorted. The issue is a bit more subtle than I thought initially.

Using `"\u00D7"` (in double quotes and no prefix) will result in [implementation-defined narrow multibyte string literal](https://en.cppreference.com/w/cpp/language/string_literal#Notes) and hence encoding of the output will be implementation-defined. To force UTF-8/16/32 we need to use `u8`/`u`/`U` prefix respectively, it will enable the "magic conversion" of `\unnnn` [escape sequences](https://en.cppreference.com/w/cpp/language/escape#Notes) to a desired encoding.

I assume implementation-defined output of the library may be a good thing. Someone may want to use single-byte national character sets. So it is up to @mpusz to decide if the library should keep encoding choice to the user or force UTF-X output. Tests will need to be adjusted afterwards.

Reply:
As of now (even in C++20) we do not have any means to print UTF-8 encoded strings to the terminal. Moreover, in C++20 we will get a new fundamental type called `char8_t` that will only work with UTF-8 strings. `char` will continue to be as it is now, mixing ASCII and implementation-defined encoding.